src/codeu/chat/util/Tokenizer.java

```diff
@@ -0,0 +1,68 @@
+
+package codeu.chat.util;
+
+import java.io.IOException;
+
+
+public final class Tokenizer{
+  private StringBuilder token;
+  private String source;
+  private int at;
+  
+  
+  public Tokenizer (String source){
```

Comment:
You'll need to keep a reference to the source string inside the constructor, something like
`this.source = source;`

The StringBuilder 'token' also needs to be initialized in this constructor